src/Microsoft.DotNet.VersionTools/Automation/GitHubHttpClient.cs

```diff
@@ -0,0 +1,100 @@
+        {
+            DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+            DefaultRequestHeaders.Add("Authorization", $"token {auth.AuthToken}");
+            DefaultRequestHeaders.Add("User-Agent", auth.User);
+            _auth = auth;
+        }
+
+        public async Task<HttpResponseMessage> PutGitHubFileAsync(
+            string fileUrl,
+            string commitMessage,
+            string newFileContents)
+        {
+            Trace.TraceInformation($"Getting the 'sha' of the current contents of file '{fileUrl}'");
+
+            string currentFile = await GetStringAsync(fileUrl);
+            string currentSha = JObject.Parse(currentFile)["sha"].ToString();
+
+            Trace.TraceInformation($"Got 'sha' value of '{currentSha}'");
+
+            Trace.TraceInformation($"Request to update file '{fileUrl}' contents to:");
+            Trace.TraceInformation(newFileContents);
+
+            string updateFileBody = JsonConvert.SerializeObject(new
+            {
+                message = commitMessage,
+                committer = new
+                {
+                    name = _auth.User,
+                    email = _auth.Email
+                },
+                content = ToBase64(newFileContents),
+                sha = currentSha
+            }, Formatting.Indented);
+
+            var bodyContent = new StringContent(updateFileBody);
+            using (HttpResponseMessage response = await PutAsync(fileUrl, bodyContent))
+            {
+                response.EnsureSuccessStatusCode();
+                Trace.TraceInformation("Updated the file successfully.");
+
+                return response;
+            }
+        }
+
+        public async Task<HttpResponseMessage> PostGitHubPullRequestAsync(
+            string title,
+            string description,
+            string originOwner,
+            string originBranch,
+            string upstreamOwner,
+            string upstreamBranch,
+            string project)
+        {
+            string createPrBody = JsonConvert.SerializeObject(new
+            {
+                title = title,
+                body = description,
+                head = $"{originOwner}:{originBranch}",
+                @base = upstreamBranch
+            }, Formatting.Indented);
+
+            string pullUrl = $"https://api.github.com/repos/{upstreamOwner}/{project}/pulls";
+
+            var bodyContent = new StringContent(createPrBody);
+            using (HttpResponseMessage response = await PostAsync(pullUrl, bodyContent))
+            {
+                response.EnsureSuccessStatusCode();
+
+                Trace.TraceInformation($"Created pull request.");
+
+                JObject responseContent = JObject.Parse(await response.Content.ReadAsStringAsync());
+                string htmlUrl = responseContent["html_url"].ToString();
+
+                Trace.TraceInformation($"Pull request page link: {htmlUrl}");
+
+                return response;
```

Comment:
Fixed by removing the return--don't know what I was thinking, nothing uses it.

---
In reply to: [69951596](https://github.com/dotnet/buildtools/pull/863#discussion_r69951596) [](ancestors = 69951596)